metricbeat/module/system/socket_summary/sockstat_linux_test.go

```diff
@@ -0,0 +1,36 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// +build linux
+
+package socket_summary
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSockstat(t *testing.T) {
+	out, err := parseSockstat("./_meta/testdata/sockstat")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, out.TCPMem, 1)
+	assert.Equal(t, out.TCPOrphan, 0)
```

Comment:
Add also a check for the UDP metric here, or compare the output with a whole expected struct, so we don't forget to modify the test if we add more metrics. This may also help to investigate the issue you are having with this metric.